REGRESSION(252485@main): [GTK] webkit_web_context_get_default() crashes in Eclipse since webkit-gtk v2.36.5, v2.36.4 was fine

mcatanzaro opened the pull request:
#### a6277d4834cce0dea2f999d4c06ffa57abfbb82d
<pre>
REGRESSION(252485@main): [GTK] webkit_web_context_get_default() crashes in Eclipse since webkit-gtk v2.36.5, v2.36.4 was fine
<a href="https://bugs.webkit.org/show_bug.cgi?id=243401">https://bugs.webkit.org/show_bug.cgi?id=243401</a>

Reviewed by Adrian Perez de Castro.

Turns out WebKit&apos;s &quot;main thread&quot; may not actually be the real main
thread. This is OK as long as it matches the GTK &quot;main thread,&quot; and as
long as the application is careful to iterate the default main context
only on the WebKit/GTK &quot;main thread,&quot; as as long as the application does
not ever attempt to use these libraries on any other thread.

The motivation to do this is if the programming language controls the
real thread 1, as is the case with Java, where the Java main thread that
applications can use to run GTK and WebKit is apparently not the same as
the real main thread that&apos;s running the JVM. These applications have no
control over what their &quot;main thread&quot; is, and it seems unkind to break
them.

I&apos;ve checked in with the GTK developers, and consensus is that this
is actually expected to work, so let&apos;s not break it.

* Source/WTF/wtf/generic/MainThreadGeneric.cpp:
(WTF::initializeMainThreadPlatform):
(WTF::isMainThread):

Canonical link: <a href="https://commits.webkit.org/253010@main">https://commits.webkit.org/253010@main</a>
</pre>

aperezdc reviewed (approved):
Ooof, indeed one can change which thread is the one where WebKit stuff runs, and as long as one uses the API always from that thread things did work&mdash;and should continue to do so.

mcatanzaro committed:
REGRESSION(252485@main): [GTK] webkit_web_context_get_default() crashes in Eclipse since webkit-gtk v2.36.5, v2.36.4 was fine
https://bugs.webkit.org/show_bug.cgi?id=243401

Reviewed by Adrian Perez de Castro.

Turns out WebKit's "main thread" may not actually be the real main
thread. This is OK as long as it matches the GTK "main thread," and as
long as the application is careful to iterate the default main context
only on the WebKit/GTK "main thread," as as long as the application does
not ever attempt to use these libraries on any other thread.

The motivation to do this is if the programming language controls the
real thread 1, as is the case with Java, where the Java main thread that
applications can use to run GTK and WebKit is apparently not the same as
the real main thread that's running the JVM. These applications have no
control over what their "main thread" is, and it seems unkind to break
them.

I've checked in with the GTK developers, and consensus is that this
is actually expected to work, so let's not break it.

* Source/WTF/wtf/generic/MainThreadGeneric.cpp:
(WTF::initializeMainThreadPlatform):
(WTF::isMainThread):

Canonical link: https://commits.webkit.org/253010@main
